### PR TITLE
Refactor CloudFlare lambda code.

### PR DESCRIPTION
**Pull request by waab76**
This change refactors the CloudFlare lambda code to (hopefully) improve separation of concerns as well as readability and testability.

### CODE REVIEW

**Comment by waab76** [comment]
Verified Lambda functionality by executing in AWS with running Graylog instance.

File: src/main/java/org/graylog/integrations/s3/GraylogS3Function.java [hunk 1]
```diff
@@ -96,118 +100,21 @@ private void setLoggerLevel() {
...
-                reader.close();
-            } catch (IOException e) {
-                // Suppress exception. Nothing can be done.
-                LOG.error("Failed to close stream for object [{}].", s3Object.getKey());
-            }
-        }
+    private GelfTransport buildGelfTransport() {
+        final GelfConfiguration gelfConfiguration = new GelfConfiguration(config.getGraylogHost(),
+                config.getGraylogPort())
+                .transport(config.getProtocolType().getGelfTransport())
```
[the remark this answers is not in the document]
Comment (lingpri) [reply src/main/java/org/graylog/integrations/s3/GraylogS3Function.java]: @waab76 - Its tcp.

File: src/test/java/org/graylog/integrations/s3/S3ScannerFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package org.graylog.integrations.s3;
+
+public class S3ScannerFactoryTest {
```
Comment (danotorrey) [inline src/test/java/org/graylog/integrations/s3/S3ScannerFactoryTest.java]: Should this test class be removed until tests are added?

File: src/test/java/org/graylog/integrations/s3/codec/S3CodecFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+@RunWith(MockitoJUnitRunner.class)
+public class S3CodecFactoryTest {
+
+    // Code Under Test
+    @InjectMocks
+    private S3CodecFactory cut;
+
+    // Mocks
+    @Mock Configuration mockConfig;
+    @Mock ObjectMapper mockObjectMapper;
```
Comment (danotorrey) [inline src/test/java/org/graylog/integrations/s3/codec/S3CodecFactoryTest.java]: Unused `ObjectMapper`
```suggestion
```

File: src/test/java/org/graylog/integrations/s3/codec/ApplicationJsonCodecTest.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+@RunWith(MockitoJUnitRunner.class)
+public class ApplicationJsonCodecTest {
+
+    // Code Under Test
+    @InjectMocks
+    private ApplicationJsonCodec cut;
+
+    // Mocks
+    @Mock
+    Configuration mockConfig;
```
Comment (danotorrey) [inline src/test/java/org/graylog/integrations/s3/codec/ApplicationJsonCodecTest.java]: I believe this unused field can be removed.
```suggestion
```

File: pom.xml [hunk 5]
```diff
@@ -88,6 +88,18 @@
...
+       <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-all</artifactId>
+            <version>1.10.19</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
```
Comment (danotorrey) [inline pom.xml]: A slightly newer version is available (if suitable).
```suggestion
            <version>4.13</version>
```

File: src/test/java/org/graylog/integrations/s3/codec/PlainTextCodecTest.java [hunk 6]
```diff
@@ -0,0 +1,4 @@
+package org.graylog.integrations.s3.codec;
+
+public class PlainTextCodecTest {
```
[the remark this answers is not in the document]
Comment (waab76) [reply src/test/java/org/graylog/integrations/s3/codec/PlainTextCodecTest.java]: Will remove.

File: src/test/java/org/graylog/integrations/s3/codec/S3CodecFactoryTest.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        given(mockConfig.getContentType()).willReturn(contentType);
+    }
+
+    // WHENs
+    private void whenGetCodecIsCalled() {
+        codec = cut.getCodec();
+    }
+
+    // THENs
+    private void thenCodecInstanceOf() {
```
[the remark this answers is not in the document]
Comment (waab76) [reply src/test/java/org/graylog/integrations/s3/codec/S3CodecFactoryTest.java]: Oops, totally meant to implement that method and use it throughout.  Doing that now.

File: src/test/java/org/graylog/integrations/s3/S3ScannerFactoryTest.java (hunk 2 above)
Comment (waab76) [reply src/test/java/org/graylog/integrations/s3/S3ScannerFactoryTest.java]: Agreed.  Will remove.

File: src/test/java/org/graylog/integrations/s3/codec/S3CodecFactoryTest.java (hunk 3 above)
Comment (waab76) [reply src/test/java/org/graylog/integrations/s3/codec/S3CodecFactoryTest.java]: I thought this was needed in order for the @InjectMocks annotation to work. Today I Learned that if an appropriate mock object isn't found, @InjectMocks will just inject a null, which is fine for this test.

File: pom.xml (hunk 5 above)
Comment (waab76) [reply pom.xml]: Indeed.

File: src/test/java/org/graylog/integrations/s3/codec/ApplicationJsonCodecTest.java (hunk 4 above)
Comment (waab76) [reply src/test/java/org/graylog/integrations/s3/codec/ApplicationJsonCodecTest.java]: Indeed it can.

**Comment by waab76** [comment]
> I really like the improvements to remove the business logic from `GraylogS3Function` 👍 What do you think about introducing a simple dependency injection tool in this PR? Too much right now, or would that clean up the `GraylogS3Function` delegation more? Completely up to you.

I debated adding dependency injection as part of this.  I was a little afraid it would be too much all in one go.  I was also a bit hesitant because I'm not really familiar with the company conventions around dependency injection.  But I think it would be a good change, so I'll give it a shot.

**Pull request merged**